### PR TITLE
Improve Detector, Dashboard Group and Dashboard checking and incorrect structure for Detector viz options

### PR DESCRIPTION
**Pull request by cory-signalfx**
# Summary

Fixes a bad type in a detector's visualization options and verifies status codes in all detector methods.

# Motivation

* The docs for `visualizationOptions` are incorrect, saying the type is an array. This fixes the OpenAPI-generated code.
* Various noun methods were not checking status codes, which meant they fell through and failed unmarshaling which is lame.

This fixes the above and adds tests to verify them.

### CODE REVIEW

File: detector_test.go [hunk 1]
```diff
@@ -33,26 +46,56 @@ func TestDeleteDetector(t *testing.T) {
...
+
+	err := client.DeleteDetector("example")
+	assert.Error(t, err, "Should have gotten an error from a missing delete")
+}
+
 func TestDisableDetector(t *testing.T) {
 	teardown := setup()
 	defer teardown()
 
-	mux.HandleFunc("/v2/detector/string/disable", verifyRequest(t, "PUT", http.StatusOK, nil, ""))
```
Comment (cory-signalfx) [inline detector_test.go]: This was actually wrong. :)

File: detector_test.go [hunk 2]
```diff
@@ -33,26 +46,56 @@ func TestDeleteDetector(t *testing.T) {
...
 	err := client.DisableDetector("string")
-	assert.NoError(t, err, "Unexpected error enabling detector")
+	assert.Error(t, err, "Should have gotten an error from a missing disable")
 }
 
 func TestEnableDetector(t *testing.T) {
 	teardown := setup()
 	defer teardown()
 
-	mux.HandleFunc("/v2/detector/string/enable", verifyRequest(t, "PUT", http.StatusOK, nil, ""))
```
Comment (cory-signalfx) [inline detector_test.go]: This was also wrong.

File: dashboardgroup.go [hunk 3]
```diff
@@ -29,6 +29,10 @@ func (c *Client) CreateDashboardGroup(dashboardGroupRequest *dashboard_group.Cre
 	}
 	defer resp.Body.Close()
 
+	if resp.StatusCode != http.StatusOK {
+		return nil, errors.New("Unexpected status code: " + resp.Status)
```
Comment (keitwb) [inline dashboardgroup.go]: Consider whether the response body might be useful to dump into the error message as well.  I guess it depends on what kinds of input might trigger a non-200 response and how helpful the response is in diagnosing it.

File: detector.go [hunk 4]
```diff
@@ -27,6 +29,11 @@ func (c *Client) CreateDetector(detectorRequest *detector.CreateUpdateDetectorRe
 	}
 	defer resp.Body.Close()
 
+	if resp.StatusCode != http.StatusOK {
+		message, _ := ioutil.ReadAll(resp.Body)
+		return nil, fmt.Errorf("Bad status %d: %s", resp.StatusCode, message)
```
Comment (keitwb) [inline detector.go]: Oh ok, you did the response body here, so I guess you know when it is useful or not.

File: dashboardgroup.go (hunk 3 above)
Comment (cory-signalfx) [reply dashboardgroup.go]: No reason to not be consistent, though. Excellent point! Will add and merge.

**Pull request merged**